### PR TITLE
Update deploy E2E test setup

### PR DESCRIPTION
**Pull request by ijjk**
Fixes E2E deploy tests as they were failing from the swc dev binary not being present for jest transpiling and skips some tests that need to be investigated/aren't expected to be working currently

### CODE REVIEW

File: run-tests.js [hunk 1]
```diff
@@ -210,7 +210,12 @@ async function main() {
     )
   })
 
-  if ((testType && testType !== 'unit') || hasIsolatedTests) {
+  if (
+    (testType &&
+      testType !== 'unit' &&
+      !process.env.NEXT_TEST_MODE === 'deploy') ||
```
Comment (almeidx) [inline run-tests.js:216]: Were you looking for `!==` instead? Otherwise, this will be comparing a boolean value to a string

Comment (ijjk) [reply run-tests.js]: Yup typed too fast here, good catch

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 756bd870456cd00fb8819cd94d0eec290f2bb628

`yarn testheadless test/e2e/i18n-api-support/index.test.ts` 
- i18n API support > should fallback rewrite non-matching API request

<details>
<summary>Expand output</summary>

  ● i18n API support › should fallback rewrite non-matching API request

    expect(received).toBe(expected) // Object.is equality

    Expected: 200
    Received: 500

      67 |       for (const path of paths) {
      68 |         const res = await fetchViaHTTP(next.url, path)
    > 69 |         expect(res.status).toBe(200)
         |                            ^
      70 |         expect(await res.text()).toContain('Example Domain')
      71 |       }
      72 |     })

      at Object.<anonymous> (e2e/i18n-api-support/index.test.ts:69:28)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/e2e-deploy | Change |
| - | - | - | - |
| buildDuration | 23.3s | 23.2s | -99ms |
| buildDurationCached | 8.7s | 8.8s | ⚠️ +71ms |
| nodeModulesSize | 479 MB | 479 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | vercel/next.js fix/e2e-deploy | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 6.037 | 5.97 | -0.07 |
| / avg req/sec | 414.08 | 418.76 | +4.68 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 2.257 | 2.163 | -0.09 |
| /error-in-render avg req/sec | 1107.59 | 1156 | +48.41 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/e2e-deploy | Change |
| - | - | - | - |
| 925.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42 kB | 42 kB | ✓ |
| main-HASH.js gzip | 29.2 kB | 29.2 kB | ✓ |
| webpack-HASH.js gzip | 1.54 kB | 1.54 kB | ✓ |
| Overall change | 73 kB | 73 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/e2e-deploy | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/e2e-deploy | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.36 kB | 1.36 kB | ✓ |
| _error-HASH.js gzip | 193 B | 193 B | ✓ |
| amp-HASH.js gzip | 308 B | 308 B | ✓ |
| css-HASH.js gzip | 327 B | 327 B | ✓ |
| dynamic-HASH.js gzip | 2.71 kB | 2.71 kB | ✓ |
| head-HASH.js gzip | 359 B | 359 B | ✓ |
| hooks-HASH.js gzip | 920 B | 920 B | ✓ |
| image-HASH.js gzip | 5.73 kB | 5.73 kB | ✓ |
| index-HASH.js gzip | 263 B | 263 B | ✓ |
| link-HASH.js gzip | 2.65 kB | 2.65 kB | ✓ |
| routerDirect..HASH.js gzip | 320 B | 320 B | ✓ |
| script-HASH.js gzip | 391 B | 391 B | ✓ |
| withRouter-HASH.js gzip | 318 B | 318 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 16 kB | 16 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/e2e-deploy | Change |
| - | - | - | - |
| _buildManifest.js gzip | 458 B | 458 B | ✓ |
| Overall change | 458 B | 458 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/e2e-deploy | Change |
| - | - | - | - |
| index.html gzip | 533 B | 533 B | ✓ |
| link.html gzip | 547 B | 547 B | ✓ |
| withRouter.html gzip | 529 B | 529 B | ✓ |
| Overall change | 1.61 kB | 1.61 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Default Build with SWC</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/e2e-deploy | Change |
| - | - | - | - |
| buildDuration | 26.1s | 26.3s | ⚠️ +190ms |
| buildDurationCached | 9.1s | 8.7s | -393ms |
| nodeModulesSize | 479 MB | 479 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | vercel/next.js fix/e2e-deploy | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 6.046 | 5.879 | -0.17 |
| / avg req/sec | 413.53 | 425.24 | +11.71 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 2.354 | 2.15 | -0.2 |
| /error-in-render avg req/sec | 1062.19 | 1162.87 | +100.68 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/e2e-deploy | Change |
| - | - | - | - |
| 925.HASH.js gzip | 178 B | 178 B | ✓ |
| framework-HASH.js gzip | 42.7 kB | 42.7 kB | ✓ |
| main-HASH.js gzip | 29.7 kB | 29.7 kB | ✓ |
| webpack-HASH.js gzip | 1.54 kB | 1.54 kB | ✓ |
| Overall change | 74.1 kB | 74.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/e2e-deploy | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/e2e-deploy | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.35 kB | 1.35 kB | ✓ |
| _error-HASH.js gzip | 179 B | 179 B | ✓ |
| amp-HASH.js gzip | 311 B | 311 B | ✓ |
| css-HASH.js gzip | 324 B | 324 B | ✓ |
| dynamic-HASH.js gzip | 2.89 kB | 2.89 kB | ✓ |
| head-HASH.js gzip | 357 B | 357 B | ✓ |
| hooks-HASH.js gzip | 920 B | 920 B | ✓ |
| image-HASH.js gzip | 5.82 kB | 5.82 kB | ✓ |
| index-HASH.js gzip | 261 B | 261 B | ✓ |
| link-HASH.js gzip | 2.78 kB | 2.78 kB | ✓ |
| routerDirect..HASH.js gzip | 322 B | 322 B | ✓ |
| script-HASH.js gzip | 392 B | 392 B | ✓ |
| withRouter-HASH.js gzip | 317 B | 317 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 16.3 kB | 16.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/e2e-deploy | Change |
| - | - | - | - |
| _buildManifest.js gzip | 457 B | 457 B | ✓ |
| Overall change | 457 B | 457 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/e2e-deploy | Change |
| - | - | - | - |
| index.html gzip | 531 B | 531 B | ✓ |
| link.html gzip | 545 B | 545 B | ✓ |
| withRouter.html gzip | 527 B | 527 B | ✓ |
| Overall change | 1.6 kB | 1.6 kB | ✓ |

</details>

</details>
Commit: 51662ff0519bdd8aeddb95dd548ac031c59db8be

File: test/e2e/i18n-data-fetching-redirect/index.test.ts [hunk 2]
```diff
@@ -7,6 +7,12 @@ import webdriver from 'next-webdriver'
 describe('i18n-data-fetching-redirect', () => {
   let next: NextInstance
 
+  // TODO: investigate tests failures on deploy
```
Comment (styfle) [inline test/e2e/i18n-data-fetching-redirect/index.test.ts:10]: Do you want to merge this?

File: test/lib/e2e-utils.ts [hunk 3]
```diff
@@ -46,7 +46,9 @@ if (testModeFromFile === 'e2e') {
   const validE2EModes = ['dev', 'start', 'deploy']
 
   if (!process.env.NEXT_TEST_JOB && !testMode) {
-    console.warn('Warn: no NEXT_TEST_MODE set, using default of start')
+    require('console').warn(
```
Comment (styfle) [inline test/lib/e2e-utils.ts:49]: Whats the purpose of require() here?

Comment (ijjk) [reply test/lib/e2e-utils.ts:49]: it bypasses jest delaying the logs from being emitted, x-ref: https://github.com/vercel/next.js/pull/35507

File: test/e2e/i18n-data-fetching-redirect/index.test.ts (hunk 2 above)
Comment (ijjk) [reply test/e2e/i18n-data-fetching-redirect/index.test.ts:10]: Yeah this will need to be investigated separately

**Review by styfle — APPROVED** [review]

**Pull request merged**